conans/test/functional/scm_test.py

```diff
@@ -168,6 +168,65 @@ def test_install_checked_out(self):
         client2.run("install lib/0.1@lasote/channel --build")
         self.assertIn("My file is copied", client2.out)
 
+    def test_submodule(self):
+        tmp = '''
+from conans import ConanFile, tools
+
+class ConanLib(ConanFile):
+    name = "lib"
+    version = "0.1"
+    scm = {{
+        "type": "git",
+        "url": "{url}",
+        "revision": "{revision}",
+        "submodule": "{submodule}"
+    }}
+'''
+        subsubmodule, _ = create_local_git_repo({"subsubmodule": "contents"})
+        submodule, _ = create_local_git_repo({"submodule": "contents"}, submodules=[subsubmodule])
+        path, commit = create_local_git_repo({"myfile": "contents"}, branch="my_release", submodules=[submodule])
+
+        def _relative_paths(folder):
+            submodule_path = os.path.join(
+                folder, 
+                os.path.basename(os.path.normpath(submodule)))
+            subsubmodule_path = os.path.join(
+                submodule_path, 
+                os.path.basename(os.path.normpath(subsubmodule)))
+            return submodule_path, subsubmodule_path
+
+        # Check old (default) behaviour
+        conanfile = tmp.format(url=path, revision=commit, submodule="none")
```

Comment:
I think this ``"none"`` evidence a possible source of user UI errors. I would suggest:
- Having not defined values "shallow, recursive", is an error and will raise an Exception. This avoids users figuring out why it doesnt work if they do a typo or something.
- This line here will skip the "submodule" field completely, not use a "None" value.

Reply:
Yeah, I think I derped here a bit. "none" here was meaning just random value, not a "shallow, recursive" which is strange behavior. I will solidify that part later! Thanks.